### ISSUE TITLE
Renaming via Zotfile causes persistent pop-ups that start piling up

### ISSUE THREAD

**Issue by ReaderGuy42** [jlegewie/zotfile#646]
I'm trying to set up Zotfile to rename my files, as I'd like to be able to use it to batch rename files (which I don't think vanilla Zotero can do?) but it's not working as intended.

a) I don't think it's actually renaming the file.

b) It's causing pop-ups that do NOT disappear even when Zotero is closed. These have been open for several minutes.

c) I can now no longer open the ZotFile plugin Preferences.

Any ideas?

Thanks!

![image](https://user-images.githubusercontent.com/71895400/224489670-02e571e7-56a2-49e3-911f-aaa93c4216ce.png)

**Comment by celtickuma** [comment]
Vanilla Zotero can batch rename files based on item metadata.
If you open an item and click on the pdf file or snapshot file (or both) and right click, you'll find "rename files from parent metadata" in the menu.  The files will be renamed based upon the naming convention which you have selected.
You can choose several files and rename them in one action, by opening the items (use the + key) and select clicking the files you wish to rename.  There is an upper limit on the number of file which can be batch renamed, so if the menu selection does not show up, drop a couple off of the list.  I tend to do them in batches of known size, so I don't hit that limit.  
You can't click on the parent item and rename the files, as you've found.
Sorry this doesn't help you with zotfile, but it should take care of your renaming task.

**Comment by ReaderGuy42** [comment]
Yeah, thanks, I've figured that out in the mean time. Originally I was struggling with this because there were a handful of sneaky top-level PDFs with no parent items which meant that if I selected all PDFs I couldn't "Rename from Metadata". Once I created parent items for all of those orphan files, it was no problem.

Nonetheless, the function still seems to be broken for Zotfile, no?
